src/features/game/expansion/Game.tsx

```diff
@@ -80,12 +83,19 @@ export const Game: React.FC = () => {
     };
   }, []);
 
+  const loadingSession =
+    gameState.matches("loading") &&
+    gameState.context.sessionId === INITIAL_SESSION;
+
   return (
     <>
       <ToastManager />
 
-      <Modal show={SHOW_MODAL[gameState.value as StateValues]} centered>
-        <Panel className="text-shadow">
+      <Modal
+        show={SHOW_MODAL[gameState.value as StateValues] && !loadingSession}
```

Comment:
Instead of rendering this tree with some conditionals could we short circuit above and return early? It may help make it easier to read

E.g.

```
  const loadingSession =
    gameState.matches("loading") &&
    gameState.context.sessionId === INITIAL_SESSION;

if (loadingSession) {
    return <div className="h-s......
}

return (
   <>
      <ToastManager />
      ...
```

Reply:
100%! This was actually what I had but then got stuck on the dumbest bug that lead to many code changes 😅 .  Will update.